File: solutions/ruby/raindrops/2/raindrops.rb
class Raindrops
  RAINDROPS = {3 => "Pling", 5 => "Plang", 7 => "Plong"}
  
  def self.convert(number)
    result = RAINDROPS.each_with_object('') do |(key, value), acc|
      acc << value if number % key == 0
    end

    result.empty? ? number.to_s : result
  end
end

File: solutions/javascript/raindrops/2/raindrops.js
//
// This is only a SKELETON file for the 'Raindrops' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const DROPS = {
  3: 'Pling',
  5: 'Plang',
  7: 'Plong'
}
export const convert = num => {
  return Object.keys(DROPS)
          .filter(key => num % key === 0)
          .map(key => DROPS[key])
          .join('') || num.toString();
}